packages/flutter_tools/lib/src/version.dart

```diff
@@ -37,11 +37,12 @@ class FlutterVersion {
     return getBranchName() == 'stable';
   }
 
-  static const Set<String> officialChannels = <String>{
-    'master',
-    'dev',
-    'beta',
-    'stable',
+  /// This maps the officialChannels ordered by stability.
+  static const Map<String, int> officialChannels = <String, int>{
```

Comment:
I don't think we need to change this to a map. I would use [List.indexOf](https://api.dartlang.org/stable/2.4.0/dart-core/List/indexOf.html) to do the lookup since it is such a small list.